Configuration/TCA/Overrides/tt_content.php

```diff
@@ -1,134 +1,65 @@
-\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin( 'Aimeos.aimeos', 'account-profile', 'Aimeos Shop - Account profile' );
-
-$GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['aimeos_account-review'] = 'pi_flexform';
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue( 'aimeos_account-review', 'FILE:EXT:aimeos/Configuration/FlexForms/AccountReview.xml' );
-\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin( 'Aimeos.aimeos', 'account-review', 'Aimeos Shop - Account review' );
-
-$GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['aimeos_account-subscription'] = 'pi_flexform';
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue( 'aimeos_account-subscription', 'FILE:EXT:aimeos/Configuration/FlexForms/AccountSubscription.xml' );
-\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin( 'Aimeos.aimeos', 'account-subscription', 'Aimeos Shop - Account subscriptions' );
-
-$GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['aimeos_account-watch'] = 'pi_flexform';
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue( 'aimeos_account-watch', 'FILE:EXT:aimeos/Configuration/FlexForms/AccountWatch.xml' );
-\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin( 'Aimeos.aimeos', 'account-watch', 'Aimeos Shop - Account watch list' );
-
-
-?>
\ No newline at end of file
+$aimeosPlugins = [
+	'Tools' => [
+		'jsonapi',
+		'locale-select'
+	],
+	'Catalog' => [
+		'catalog-attribute',
+		'catalog-count',
+		'catalog-detail',
+		'catalog-filter',
+		'catalog-home',
+		'catalog-list',
+		'catalog-price',
+		'catalog-search',
+		'catalog-session',
+		'catalog-stage',
+		'catalog-stock',
+		'catalog-suggest',
+		'catalog-supplier',
+		'catalog-tree',
+	],
+	'Basket' => [
+		'basket-bulk',
+		'basket-related',
+		'basket-small',
+		'basket-standard',
+	],
+	'Checkout' => [
+		'checkout-confirm',
+		'checkout-standard',
+		'checkout-update',
+	],
+	'Account' => [
+		'account-download',
+		'account-history',
+		'account-favorite',
+		'account-profile',
+		'account-review',
+		'account-subscription',
+		'account-watch',
+	]
+];
+
+foreach ( $aimeosPlugins as $groupName => $pluginGroup ) {
+	foreach ( $pluginGroup as $pluginName ) {
+		$GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['aimeos_' . $pluginName] = 'pi_flexform';
+		\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue(
+			'aimeos_' . $pluginName,
+			'FILE:EXT:aimeos/Configuration/FlexForms/' . str_replace( '-', '', ucwords( $pluginName, '-' ) ) . '.xml'
+		);
+		\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin(
+			'Aimeos.aimeos',
+			$pluginName,
+			'LLL:EXT:aimeos/Resources/Private/Language/admin.xlf:aimeos_' . $pluginName . '.title',
+			'aimeos_'  .$pluginName,
+			'Aimeos Shop: ' .$groupName
```

Comment:
According to https://docs.typo3.org/m/typo3/book-extbasefluid/master/en-us/4-FirstExtension/7-configuring-the-plugin.html, registerPlugin() only has four parameters

Reply:
Indeed, my mistake! That feature is available in Version 10.4+ only:
https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.4/Feature-91008-ItemGroupingForTCASelectItems.html
In this case, you could prepare the translations and add a "group title" for later updates when v9 gets deprecated.

Reply:
For v10+ it's OK but shouldn't it always be 'Aimeos Shop'? With a more fine-grained grouping, there may be too many top-level entries and you also have the problem again that this value can't be translated, can it?

Reply:
Everything is translatable in this case. If a key in your XLF-file exists, it can be referenced with `LLL:EXT:aimeos/Resources/Private/Language/LANGUAGEFILE.xlf:KEY`.

But you are right, choosing "Aimeos Shop: " as prefix for all items and "Aimeos Shop" as their main group, should be enough.

Anyway, this can only be done in v10, so i will close this PR and reissue it again if support for v9 is dropped.

Reply:
...or we could just prepare the code in this direction (to minify the required lines of code) and add the grouping when it is available in v10?

Reply:
Don't know if the minification is a good idea because it introduces variables into the global namespace